src/main/java/com/iexec/core/replicate/ReplicatesService.java

```diff
@@ -464,7 +499,7 @@ public void updateReplicateStatus(OptimisticLockingFailureException exception,
                         "nextAction:{}, chainTaskId:{}, walletAddress:{}]",
                 replicate.getCurrentStatus(), newStatusCause, nextAction, chainTaskId, walletAddress);
 
-        return Optional.ofNullable(nextAction); // should we return a default action when null?
+        return Optional.ofNullable(nextAction);
     }
 
     @Recover
```

Comment:
Could the `chainTaskId`, `wallet address` and `status update` be written to the log ?
Could we have `log.error("message", exception)` in the log method instead of `printStackTrace()` ?

Reply:
I am speaking of `updateReplicateStatus(OptimisticLockingFailureException exception, ...)` and `logUpdateReplicateStatusRecover(OptimisticLockingFailureException exception)` methods

Reply:
As far as I remember, those values are displayed in the error message. I'm not sure it would add any value to duplicate them (?).
However, second point is right. See https://github.com/iExecBlockchainComputing/iexec-core/pull/492/commits/388a3d19ef57dfd745f7b5ebe6d51b5d52f26377.